internal/auth/login_handler_test.go

```diff
@@ -0,0 +1,348 @@
+	defer fakeOAuthServer.Close()
+
+	// Setup mock PDS server to handle identity resolution and OAuth discovery
+	mockPDS := setupMockPDSWithOAuth(t, "test.bsky.app", "did:plc:test123", fakeOAuthServer.URL)
+	defer mockPDS.Close()
+
+	// Create real OAuth client
+	oauthClient := testOAuthClient(t)
+	handler := setupTestLoginHandler(t, mockPDS.URL, oauthClient)
+
+	// Create request
+	req := httptest.NewRequest("GET", "/login?handle=test.bsky.app", nil)
+	w := httptest.NewRecorder()
+
+	// Execute request
+	handler.ServeHTTP(w, req)
+
+	// Verify response
+	require.Equal(t, http.StatusSeeOther, w.Code)
+	location := w.Header().Get("Location")
+	require.Contains(t, location, fakeOAuthServer.URL)
+
+	// Verify session cookies were set
+	cookies := w.Result().Cookies()
+	sessionNames := make(map[string]bool)
+	for _, cookie := range cookies {
+		sessionNames[cookie.Name] = true
+	}
+	require.True(t, sessionNames["dpop-session"], "DPoP session should be created")
+	require.True(t, sessionNames["auth-session"], "Auth session should be created")
+}
+
+func TestLoginHandler_WithPDSURL(t *testing.T) {
+	// Setup fake OAuth server
+	fakeOAuthServer := fakeAuthServer(t, nil)
+	defer fakeOAuthServer.Close()
+
+	// Setup mock PDS server to handle identity resolution and OAuth discovery
+	mockPDS := setupMockPDSWithOAuth(t, "test.bsky.app", "did:plc:test123", fakeOAuthServer.URL)
+	defer mockPDS.Close()
+
+	// Create real OAuth client
+	oauthClient := testOAuthClient(t)
+	handler := setupTestLoginHandler(t, mockPDS.URL, oauthClient)
+
+	// Create request
+	req := httptest.NewRequest("GET", "/login?handle=test.bsky.app", nil)
+	w := httptest.NewRecorder()
+
+	// Execute request
+	handler.ServeHTTP(w, req)
+
+	// Verify response
+	require.Equal(t, http.StatusSeeOther, w.Code)
+
+	// Verify redirect location and session cookies
+	location := w.Header().Get("Location")
+	require.Contains(t, location, fakeOAuthServer.URL)
+
+	cookies := w.Result().Cookies()
+	var dpopCookie *http.Cookie
+	for _, cookie := range cookies {
+		if cookie.Name == "dpop-session" {
+			dpopCookie = cookie
+			break
+		}
+	}
+	require.NotNil(t, dpopCookie, "DPoP session cookie should be set")
+	require.Equal(t, "/", dpopCookie.Path, "DPoP session cookie should have root path")
+}
+
+func TestLoginHandler_InvalidHandle(t *testing.T) {
+	// Setup fake OAuth server
+	fakeOAuthServer := fakeAuthServer(t, nil)
+	defer fakeOAuthServer.Close()
+
+	oauthClient := testOAuthClient(t)
+	handler := setupTestLoginHandler(t, "", oauthClient)
+
+	// Test with invalid handle
+	req := httptest.NewRequest("GET", "/login?handle=invalid-handle", nil)
+	w := httptest.NewRecorder()
+
+	handler.ServeHTTP(w, req)
+
+	require.Equal(t, http.StatusBadRequest, w.Code)
+	require.Contains(t, w.Body.String(), "invalid")
+}
+
+func TestLoginHandler_MissingHandle(t *testing.T) {
+	// Setup fake OAuth server
+	fakeOAuthServer := fakeAuthServer(t, nil)
+	defer fakeOAuthServer.Close()
+
+	oauthClient := testOAuthClient(t)
```

Comment:
wonder if it makes more sense to use a dummy OAuthClient here so we don't need to set up the fake server